src/ngraph/op/non_zero.hpp

```diff
@@ -42,13 +42,26 @@ namespace ngraph
                 /// \brief Constructs a NonZero operation.
                 ///
                 /// \param arg Node that produces the input tensor.
-                NonZero(const Output<Node>& arg);
+                /// \param index_element_type produce indices. Currently, only int64 or int32 are
+                ///                           supported
+                NonZero(const Output<Node>& arg, const element::Type& index_element_type);
 
                 bool visit_attributes(AttributeVisitor& visitor) override;
                 void validate_and_infer_types() override;
 
                 virtual std::shared_ptr<Node>
                     clone_with_new_inputs(const OutputVector& new_args) const override;
+
+                element::Type get_output_type() const { return m_index_element_type; }
+                void set_output_type(element::Type output_type)
+                {
+                    m_index_element_type = output_type;
+                }
+                // Overload collision with method on Node
+                using Node::set_output_type;
+
+            protected:
+                element::Type m_index_element_type;
```

Comment:
Just noticed in the specs that this should be a string. The class member can still be an enum but we should add a constructor that takes a string and converts it to enum. Refer to this PR for more details on how to do it https://github.com/NervanaSystems/ngraph/pull/4432/files?file-filters%5B%5D=.cpp#diff-b358c2a0e091411efef86dea664fb6cdR24

Reply:
Done